### DISCUSSION TITLE
Microphone selection grayed out on chrome

### DISCUSSION THREAD

**Discussion by julesvirallinen** [ffd8/P5LIVE#83] [General]
This is probably chrome specific, but raising here just in case there's an easy solution. 

Microphone support works locally and on the web version, but the microphone selection is grayed out. This happens on local, local https, production https. Also happens on my own vj-code app, so probably not specific to P5LIVE. 

I've tried the `#unsafely-treat-insecure-origin-as-secure` and everything else. I guess firefox works, but MIDI doesn't work there. 

<img width="364" alt="CleanShot 2022-12-04 at 14 06 12@2x" src="https://user-images.githubusercontent.com/4622905/205489500-bb4da056-ec3e-4080-a674-963455dc3883.png">

Anyone run into any solutions? I've eventually been able to change it at gigs after some panicking but not in any coherent way.

**Comment by ffd8** [comment]
@julesvirallinen - just seeing this now.. yeah chrome and mic selection is weird... while you can technically click the 'manage' and select another one – it doesn't usually stay set when reloading the page... it's all actually dependent on the sample-rate of your inputs within the `Audio MIDI Setup` utility. Chrome usually grabs the worst quality one (usually 44,100 khz). Soo when it's selecting my 'blackhole' input instead of mic, i just have to bump blackhole to be higher and make sure the mic is that lower amount – then refresh the page and voila, it picks the mic. I think this is in order to make sure transmission of mic in vidchat etc works smoothly.

**Reply by julesvirallinen** [reply]
Hmm interesting. I eventually just bought Audio Hijack / Loopback and never looked back
